### ISSUE TITLE
Support and propagate noexcept-qualified function types

### ISSUE THREAD

**Issue by RedBeard0531** [Naios/function2#15]
@Naios 

Basically, I want this to work
```cpp
// This works today
using yesexcept_func = fu2::function<void()>;
static_assert(!noexcept(std::declval<yesexcept_func>()()));

// This fails to compile
using noexcept_func = fu2::function<void() noexcept>;
static_assert(noexcept(std::declval<noexcept_func>()()));
```
https://godbolt.org/g/zxVvKT
Propagation may require c++17.

(Removed most of the issue template boilerplate since it doesn't seem to apply to feature requests)

**Comment by Naios** [comment]
Hey @RedBeard0531,
thank you for your feature request.
I also thought about this, and probably this will be implement in the future.
Yes you are right, it requires C++17, if you would like to open a PR,
I could provide you the necessary help.

**Comment by Naios** [comment]
Thanks for your feature request.
It was implemented by c0b8792f10 and is available now in v3.1.0.
